sys_posix.c

```diff
@@ -746,3 +746,22 @@ const char* Sys_HomeDirectory(void)
 	}
 	return "";
 }
+
+char* Sys_HardwareID(void)
+{
+        FILE *f;
+        char buffer[255] = {0};
+	buffer[0] = '0';
```

Comment:
You could do buffer[255] = {'0'}; above and the first entry would be ascii zero while the rest would be zero